### ISSUE TITLE
Attendees listing should show all attendees

### ISSUE THREAD

**Issue by jsbalsera** [AsociacionDrupalES/DrupalCampEs#80]
The attendees listing should show all the attendees. If the ticket are assigned to a different user than the buyer, the attendees listing should show the receiver user.

**Comment by DavidHernandez** [comment]
Individual sponsors are attendees too.

**Comment by DavidHernandez** [comment]
The part I reported has been fixed. The part reported by @jsbalsera is still not fixed.

**Comment by DavidHernandez** [comment]
Seems that we already have 52 registered attendees: http://2015.drupalcamp.es/es/admin/conference/2/tickets

But the view shows less than half: http://2015.drupalcamp.es/es/drupalcamp-spain-2015/asistentes

**Comment by jsbalsera** [comment]
The numbers doesn't sum up in the admin interface, neither. In the general listing that David linked there are 6 individual sponsors and 11 attendees to the business day, but if we check the actual listings there are 5 individual sponsors and 10 attendees to the business day. (I didn't count the early bird tickets, sorry)

In the case of IS I bought two tickets and I am only once there, just in case that it helps.

**Comment by eurodev** [comment]
I think we should appear only once in this list. In fact, I remember that someone from the local group fixed this issue some days ago, because I have been shown twice in this list in some instant in the past.

**Comment by jsbalsera** [comment]
Javi, but if you buy a second ticket and assign it to another user then
this other user should appear as attendee...

On Thu, Apr 16, 2015 at 12:53 PM, Javier Carranza <notifications@github.com>
wrote:

> I think we should appear only once in this list. In fact, I remember that
> someone from the local group fixed this issue some days ago, because I have
> been shown twice in this list in some instant in the past.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/AsociacionDrupalES/DrupalCampEs/issues/80#issuecomment-93706957>
> .
>



-- 
Un saludo!

**Comment by eurodev** [comment]
Ahm, ok. In that case, you are right :)

**Comment by DavidHernandez** [comment]
The view should list users who registered to the event, not the person 
who bought the ticket.

**Comment by jsbalsera** [comment]
It looks like my fork became crazy or something.

Anyway, I have being trying to change the view but couldn't find the original feature, so I went with a hook_views_query_alter. The problem is that the relationship between the users and the ticket_registration is made with the author_id (the buyer of the ticket) and not with the user_uid, and you can't create that relationship (the definition of the entity, maybe?) With this code it should work...

```php
function aedcamp_views_query_alter(&$view, &$query) {
  if ($view->name === 'aed_attendees' && $view->current_display == 'page') {
    $view->query->table_queue['ticket_registration_users']['join']->field = 'user_uid';
  }
}
```

**Comment by jsbalsera** [comment]
@DavidHernandez @fran I can't find the feature with the view. Actually, if I search the name of the view (aed_attendees) in the repository it doesn't find anything, neither...

**Comment by eurodev** [comment]
It should be also desirable to add an order criteria. I guess that now it's oredered by order_id desc, so you can't actually find anyone unless you look at all the pages.

**Comment by eurodev** [comment]
Oh, I've just realized that Jesus was already kept that in mind :-)

**Comment by jsbalsera** [comment]
Marcelo says that now the view should be exported, I'll take a look this afternoon

**Comment by jsbalsera** [comment]
Fixed :-) There is also a display for Individual Sponsors and another for Business day attendes, so #85 it's also fixed.

**Comment by DavidHernandez** [comment]
This page is still broken. At the moment, it's showing only 54 attendees, while the conferences page shows 80 tickets sold, plus 7 individual sponsors.

![wrong-attendees](https://cloud.githubusercontent.com/assets/877002/7345081/d888b454-ecd1-11e4-8ac2-054967f8514c.png)

**Comment by DavidHernandez** [comment]
My guess is that we can close this. I will create another task for the 2016 milestone, so we don't have to manually complete the payments to show the attendees.
